src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/OracleEngine.java

```diff
@@ -162,17 +226,40 @@ protected int entityToPreparedStatement(final DbEntity entity, final PreparedSta
                 }
                 switch (column.getDbColumnType()) {
                     case BLOB:
-                        ps.setBytesForBlob(i, objectToArray(val));
+                        final byte[] valArray = objectToArray(val);
+                        if (fromBatch && valArray.length > MIN_SIZE_FOR_BLOB) {
+                            // Use a blob for columns greater than 4K when inside a batch
+                            // update because the Oracle driver creates one and only releases
+                            // it when the PreparedStatement is closed. This will be closed
+                            // explicitly when the batch is flushed.
+                            final Blob blob = ps.getConnection().createBlob();
+                            blob.setBytes(1, objectToArray(val));
```

Comment:
```suggestion
                            blob.setBytes(1, valArray);
```
`objectToArray` is going to serialize _val_ again, but we already have the result of that conversion